aspnetcore/fundamentals/static-files.md

```diff
@@ -1,17 +1,1104 @@
+
+* Set `ASPNETCORE_ENVIRONMENT` in `Properties/launchSettings.json` to any value other than `Development`.
+
+* Disable static web assets by setting `<StaticWebAssetsEnabled>` to `false` in the app's project file. ***WARNING:*** Disabling static web assets disables [Razor class libraries](xref:razor-pages/ui-class).
+
+* Add the following XML to the project file:
+
+  ```xml
+  <ItemGroup>
+    <Content Remove="wwwroot\**" />
+  </ItemGroup>
+  ```
+
+The following code updates <xref:Microsoft.AspNetCore.Hosting.IWebHostEnvironment.WebRootPath%2A> to a non-Development value (`Staging`), guaranteeing duplicate content is returned from `wwwroot-custom` rather than `wwwroot`:
+
+```csharp
+var builder = WebApplication.CreateBuilder(new WebApplicationOptions
+{
+    Args = args,
+    EnvironmentName = Environments.Staging,
+    WebRootPath = "wwwroot-custom"
+});
+```
+
+When developing a server-side Blazor app and testing locally, see <xref:blazor/fundamentals/static-files#static-files-in-non-development-environments>.
+
+:::moniker-end
+
+## Serve files outside of the web root directory via `UseStaticFiles`
+
+Consider the following directory hierarchy with static files residing outside of the app's [web root](xref:fundamentals/index#web-root) in a folder named `ExtraStaticFiles`:
+
+* `wwwroot`
+  * `css`
+  * `images`
+  * `js`
+* `ExtraStaticFiles`
+  * `images`
+    * `red-rose.jpg`
+
+A request can access `red-rose.jpg` by configuring a new instance of Static File Middleware:
+
+Namespaces for the following API:
+
+```csharp
+using Microsoft.Extensions.FileProviders;
+```
+
+In the request processing pipeline after the existing call to either <xref:Microsoft.AspNetCore.Builder.StaticAssetsEndpointRouteBuilderExtensions.MapStaticAssets%2A> (.NET 9 or later) or <xref:Microsoft.AspNetCore.Builder.StaticFileExtensions.UseStaticFiles%2A> (.NET 8 or earlier):
+
+```csharp
+app.UseStaticFiles(new StaticFileOptions
+{
+    FileProvider = new PhysicalFileProvider(
+        Path.Combine(builder.Environment.ContentRootPath, "ExtraStaticFiles")),
+    RequestPath = "/static-files"
+});
+```
+
+In the preceding code, the `ExtraStaticFiles` directory hierarchy is exposed publicly via the `StaticFiles` URL segment. A request to `https://{HOST}/StaticFiles/images/red-rose.jpg`, where the `{HOST}` placeholder is the host, serves the `red-rose.jpg` file.
+
+The following markup references `ExtraStaticFiles/images/red-rose.jpg`:
+
+```html
+<img src="static-files/images/red-rose.jpg" alt="A red rose" />
+```
+
+For the preceding example, tilde-slash notation is supported in Razor Pages and MVC views (`src="~/StaticFiles/images/red-rose.jpg"`), not for Razor components in Blazor apps.
+
+## Serve files from multiple locations
+
+:::moniker range=">= aspnetcore-6.0"
+
+*The guidance in this section applies to Razor Pages and MVC apps. For guidance that applies to Blazor Web Apps, see <xref:blazor/fundamentals/static-files#serve-files-from-multiple-locations>.*
```

Comment:
Isn't the guidance the same? Can we remove this duplication?

Reply:
That's what I meant about deltas. It wasn't the same IIRC, but I'll have to look. I'll check and get back to you ... probably Tuesday morning.